code/datums/action.dm

```diff
@@ -106,28 +106,68 @@
 
 
 
-
+#define XACT_USE_INCAP		(1 << 0)
+#define XACT_USE_LYING		(1 << 1)
+#define XACT_USE_BUCKLED	(1 << 2)
+#define XACT_USE_STAGGERED	(1 << 3)
+#define XACT_USE_FORTIFIED	(1 << 4)
+#define XACT_USE_CRESTED	(1 << 5)
+#define XACT_USE_NOTTURF	(1 << 6)
 
 /datum/action/xeno_action
 	var/action_icon_state
 	var/plasma_cost = 0
 	var/mechanics_text = "This ability not found in codex." //codex. If you are going to add an explanation for an ability. don't use stats, give a very brief explanation of how to use it.
+	var/use_state_flags = NONE // bypass use limitations checked by can_use_action()
+	var/on_cooldown
 
 /datum/action/xeno_action/New(Target)
 	..()
 	button.overlays += image('icons/mob/actions.dmi', button, action_icon_state)
 
 /datum/action/xeno_action/can_use_action()
 	var/mob/living/carbon/Xenomorph/X = owner
-	if(X && !X.incapacitated() && !X.lying && !X.buckled && X.plasma_stored >= plasma_cost && !X.stagger)
-		return TRUE
+	if(!X)
+		return FALSE
```

Comment:
Add a `stack_trace("[src] action attempted to check can_use_action() without an owner")`